scripts/Jenkinsfiles/quality

```diff
@@ -0,0 +1,219 @@
+def runQualityTests() {
+    sshagent(credentials: ['jenkins-worker'], ignoreMissing: true) {
+        // Determine git refspec, branch, and clone type
+        git_shallow_clone = true
+        git_extensions = []
+        if (env.ghprbActualCommit) {
+            git_branch = "${ghprbActualCommit}"
+            refspec = "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*"
+            if (SHARD == "4") {
+                git_shallow_clone = false
+                git_extensions.add([$class: 'WipeWorkspace'])
+                refspec = refspec + " +refs/heads/master:refs/remotes/origin/master"
+            }
+        } else {
+            git_branch = 'master'
+            refspec = "+refs/heads/master:refs/remotes/origin/master"
+        }
+        git_extensions.add([$class: 'CloneOption', honorRefspec: true, noTags: true, shallow: git_shallow_clone ])
+
+        checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: git_branch]],
+            doGenerateSubmoduleConfigurations: false, extensions: git_extensions, submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'jenkins-worker',
+            refspec: refspec,
+            url: 'git@github.com:edx/edx-platform.git']]]
+
+        sh "bash scripts/all-tests.sh"
+        stash includes: '**/reports/**/*, **/reports/**/*', name: "${TEST_SUITE}-${SHARD}-reports"
+    }
+}
+
+def qualityTestCleanup() {
+    archiveArtifacts allowEmptyArchive: true, artifacts: '**/reports/**/*,test_root/log/**/*.log,*.log'
+    junit '**/reports/quality.xml, reports/quality_junitxml/*.xml'
```

Comment:
Do we actually write a `quality.xml` anywhere?  Not finding one on an initial search.